### DISCUSSION TITLE
structuring the list of layers

### DISCUSSION THREAD

**Discussion by d-v-b** [google/neuroglancer#658] [Ideas]
When working with a lot of layers, the list of layer selectors can get obtrusive and cluttered. See this example:

![image](https://github.com/user-attachments/assets/302d3ab9-046b-4a1a-b4f3-3cc622c1a5de)

First, is there a neuroglancer setting to adjust this layout? Because if so, it might make the remainder of this discussion redundant :). If not, maybe it would be good to consider some alternatives to the current layout. 

The current layout basically puts all the layers in a horizontal element, reflowing vertically as needed to keep that element inside its allotted space. This is an efficient use of space, but has some downsides:

- it's visually cluttered with a lot of layers
- layer selection requires a visual search
- there's no easy way to group or organize layers

I think some of these problems would be addressed by using a different layout for the layer selectors. The simplest idea in my mind would be a scrollable vertical element that represents the layers as a list. this list could be sorted (e.g., sort by which layers are currently visible), and it could be widened to give more summary information about individual layers (e.g., highlight layers that are under the mouse at any given time). 

I'm curious to hear other thoughts here. Unfortunately I have none of the skills required to make this change, so I don't know what the code changes would look like, which keeps this rather abstract instead of an issue or a PR. Maybe if I learn a bit more about how neuroglancer is laid out I can have a more concrete proposal.

**Comment by jbms** [comment]
There is a separate layer panel that shows the layers in a vertical list --- there is a button in the top bar for displaying that panel.  You can then use the x button to archive layers which means they are not rendered and also are excluded from the normal layer bar and display only in the separate layer panel.

I'm not sure if that addresses your concern, since it only helps if you don't need to display all of the layers.

I'm curious to know more about your use case for working with many layers, especially your example where you have many numbered layers.

**Reply by jbms** [reply]
If you have a large number of layers because your data is tiled over multiple arrays, note that you can also have multiple data sources in a single layer.

**Reply by d-v-b** [reply]
I'm working with large field of view microscopy images that are acquired as a sequence of overlapping tiles. It's useful to view each image with independent rendering settings (e.g., different display colors) so that the overlap region between adjacent tiles can be visually inspected for misalignment between the two images. It's also important to be able to quickly identify which tiles are being examined (i.e., under the cursor) at any time -- I use changes in the image intensity display to infer this. Is there a better way to do that?

As I understand it, putting all the tiles into a single layer would require that they share the same shader, which would make the above workflow impossible. Is there a way to give the sub-layers their own shaders?

And if were to represent all the tiles as a single layer, wouldn't I lose the ability to resolve which sub-layer was in the FOV at any given time?  

> There is a separate layer panel that shows the layers in a vertical list --- there is a button in the top bar for displaying that panel. You can then use the x button to archive layers which means they are not rendered and also are excluded from the normal layer bar and display only in the separate layer panel.

Thanks, I didn't know about this display. it's helpful, although it's not a replacement for the list of layers at the top, as the entries in the vertical list don't give an indication of which layers are under the cursor at any given time.
